components/chessboard/ChessBoard.tsx

```diff
@@ -316,33 +312,23 @@ function mouseDown(event: MouseEvent) {
   // console.log("pos: " + event.clientX + " " + event.clientY);
 }
 
-function mouseMove(event: MouseEvent) { }
+function mouseMove(event: MouseEvent) {}
 
-function mouseUp(event: MouseEvent) { }
+function mouseUp(event: MouseEvent) {}
 
 interface PromotionParams {
   board: Chess;
   selection: string;
   makeAmove(move: Move): void;
 }
 
-
 export default function NewBoard(props: ChessBoardProps) {
-
   const userInfo = useContext(UserInfoContext);
 
-  let light = "white";
-  let dark = "#fca311";
-
-  if (userInfo) {
-    if (userInfo.user.boardLightColor) {
-      light = userInfo.user.boardLightColor;
-    }
-    if (userInfo.user.boardDarkColor) {
-      dark = userInfo.user.boardDarkColor;
-    }
-  }
-
+  let light =
```

Comment:
I added colorscheme as an optional prop in again because otherwise to be able to play around with different colors from profile, the user would need to update the user state every time they change a color which leads to a rerender of the entire tree (from userinfoprovider and its children). Additionally, it would be bad if the user context didnt have the same value as whats stored in the db, so we'd also have to make requests to update the colors in the db to make sure the context and db are on the same page. This approach should avoid rerendering and unnecessary requests.

If no colorscheme is passed in, the user context colors take next priority, then the default if no context exists.